core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/hybrid/HybridRoleManager.java

```diff
@@ -815,24 +815,25 @@ public Map<String, List<String>> getHybridRoleListOfUsers(List<String> userNames
                         JDBCCaseInsensitiveConstants.GET_INTERNAL_ROLE_V2_LIST_OF_USERS_SQL_CASE_INSENSITIVE
                         : JDBCCaseInsensitiveConstants.GET_INTERNAL_ROLE_LIST_OF_USERS_SQL_CASE_INSENSITIVE;
             }
-            for (int i = 0; i < userNames.size(); i++) {
-
-                userNames.set(i, userNames.get(i).replaceAll("'", "''"));
-                usernameParameter.append("LOWER('").append(userNames.get(i)).append("')");
+            for (int i = 0; i < usersCount; i++) {
 
-                if (i != userNames.size() - 1) {
-                    usernameParameter.append(",");
+                dynamicString.append("LOWER(?)");
+                if (i != usersCount - 1) {
+                    dynamicString.append(",");
                 }
             }
         }
 
-        sqlStmt = sqlStmt.replaceFirst("\\?", Matcher.quoteReplacement(usernameParameter.toString()));
+        sqlStmt = sqlStmt.replaceFirst("\\?", dynamicString.toString());
         try (Connection connection = DatabaseUtil.getDBConnection(dataSource);
                 PreparedStatement prepStmt = connection.prepareStatement(sqlStmt)) {
-            prepStmt.setInt(1, tenantId);
-            prepStmt.setInt(2, tenantId);
-            prepStmt.setInt(3, tenantId);
-            prepStmt.setString(4, domainName);
+            for (int i = 0; i < usersCount; i++) {
+                prepStmt.setString(i + 1, userNames.get(i));
+            }
+            prepStmt.setInt(usersCount + 1, tenantId);
+            prepStmt.setInt(usersCount + 2, tenantId);
+            prepStmt.setInt(usersCount + 3, tenantId);
+            prepStmt.setString(usersCount + 4, domainName);
```

Comment:
```suggestion
        int index = 1;
        for (String userName : userNames) {
            prepStmt.setString(index++, userName);
        }
        prepStmt.setInt(index++, tenantId);
        prepStmt.setInt(index++, tenantId);
        prepStmt.setInt(index++, tenantId);
        prepStmt.setString(index++, domainName);
```

Reply:
A suggestion for better readability.

Reply:
Fixed with https://github.com/wso2/carbon-kernel/pull/3984/commits/f468328ba739ebaecc6540b98b421b19c5d76c8a